collect_app/src/main/java/org/odk/collect/android/application/initialization/ManagedConfigManager.kt

```diff
@@ -0,0 +1,62 @@
+package org.odk.collect.android.application.initialization
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.content.RestrictionsManager
+import android.os.Bundle
+import org.odk.collect.android.projects.ProjectCreator
+import org.odk.collect.android.projects.SettingsConnectionMatcher
+import org.odk.collect.projects.ProjectsRepository
+import org.odk.collect.settings.ODKAppSettingsImporter
+import org.odk.collect.settings.SettingsProvider
+import org.odk.collect.settings.keys.MetaKeys.KEY_INSTALL_ID
+
+/**
+ * Manages configuration changes from a mobile device management system.
+ *
+ * See android.content.APP_RESTRICTIONS in AndroidManifest for supported configuration keys.
+ */
+class ManagedConfigManager (
+    private val settingsProvider: SettingsProvider,
+    private val projectsRepository: ProjectsRepository,
+    private val projectCreator: ProjectCreator,
+    private val settingsImporter: ODKAppSettingsImporter,
+    private val context: Context
```

Comment:
Doing something quick and dirty that keeps all of the functionality contained, I'm not suggesting we'd want to pass context into something like this